samples/grpc/shopping-cart-service-java/src/main/java/shopping/cart/ShoppingCart.java

```diff
@@ -70,33 +70,22 @@ public State checkout(Instant now) {
     }
 
     public Summary toSummary() {
-      return new Summary(items, isCheckedOut());
-    }
-
-    public boolean hasItem(String itemId) {
-      return items.containsKey(itemId);
+      // filter out removed items
+      final Map<String, Integer> itemsForSummary = new HashMap<>();
+      items.forEach((itemId, quantity) -> {
+        if (quantity > 0) itemsForSummary.put(itemId, quantity);
+      });
+      return new Summary(itemsForSummary, isCheckedOut());
     }
 
     public State updateItem(String itemId, int quantity) {
-      if (quantity == 0) {
-        items.remove(itemId);
-      } else {
-        items.put(itemId, quantity);
-      }
+      int newQuantity = items.getOrDefault(itemId, 0) + quantity;
+      items.put(itemId, newQuantity);
```

Comment:
is there a reason why we put 0 items, and then filter in other places? could it remove here instead if 0?

Reply:
That was mostly to keep it roughly the same with the RES version, we could remove instead here since it is a more natural design for non RES.

Reply:
If add 1, remove 1, add 1, remove 1 wold reordered to add 1, remove 1, remove 1, add 1 we get different end results with RES?

Reply:
Back to removing non-present in grpc projection samples in [db20600](https://github.com/akka/akka-projection/pull/876/commits/db20600470cff61c8f203f1d817b6bce204786a3)